### ISSUE TITLE
No Transcript on Chrome When Upgraded to MacOS Monterey on MacBook Pro

### ISSUE THREAD

**Issue by DeanWronowski** [speechly/speechly#77]
### Describe the bug
Everything has been working perfectly until I decided to upgrade Mac OS to Monterey on my Macbook Pro.
A more detailed explanation can be found here https://github.com/speechly/speech-recognition-polyfill/issues/10

### To Reproduce
Load https://speechly-demos.herokuapp.com/flight-booking in Chrome on a MacBook Pro running Mac OS Monterey

### Expected behaviour
Transcript to be displayed

### Environment
Platform: Desktop
OS: Mac OS Monterey
Browser Chrome
Chrome Version: Version 95.0.4638.69 (Official Build) (x86_64)

**Comment by aukkonen** [comment]
Hi @DeanWronowski 

I'm unable to reproduce this on
MacOS Monterey Version 12.0.1 (21A559) and
Chrome version 95.0.4638.69 (Official Build) (x86_64).

I tried both the Flight Booking Demo as well as a locally running version of the browser-client-example.

What version of Monterey are you on?

**Comment by DeanWronowski** [comment]
macOS Monterey Version 12.0.1
Processor: 2.6GHz 6-core Intel Core i7
Chrome: Version 95.0.4638.69 (Official Build) (x86_64)

I've disabled all the plugins in Chrome as well as the Antivirus.

**Comment by aukkonen** [comment]
@DeanWronowski 
We have seen similar issues in the past, but they have been notoriously difficult to reproduce. Have you tried to restart Chrome?

**Comment by DeanWronowski** [comment]
@aukkonen 
Yeah, restarted the machine as well as Chrome. Doesn't make any difference. I'm more than happy to go on Zoom or Hangouts if that makes it easier?

Is it to do with the data not sending up correctly? More than happy to give provide more data where necessary?

**Comment by aukkonen** [comment]
@DeanWronowski 
I  checked the data we had received for the example that you shared the screenshot of (https://user-images.githubusercontent.com/3969540/141072912-c51acfaf-b8c0-44b2-ba78-c0eac68abf05.png), and it looks like there's only silence. That is, we did receive audio samples for about 3 seconds but they are all zero.

We'll investigate this further, and I'll get back to you. Sorry for the inconvenience!

**Comment by DeanWronowski** [comment]
No worries @aukkonen  - give me a shout if you need anything.

**Comment by aukkonen** [comment]
@DeanWronowski 
Have you tried to reset Chrome's microphone permission on sites that use the browser-client? You can do this e.g. for the Flight Booking demo by opening https://speechly-demos.herokuapp.com/flight-booking, clicking on the "lock" icon to the left of the URL, and then selecting "Reset permission". Reload the site, and it should ask for the mic permission.

I'm not sure if this would do it, but Safari and Firefox tend to re-acquire the mic permission each time you load the page, while Chrome likes to persist the choice. It might be worth a try.

**Comment by DeanWronowski** [comment]
Okay, here's what I did.
I reset the permissions, reloaded the page, pressed the microphone icon, accepted the permissions, talked and then nothing happens. I even pressed and hold the button for a few minutes but still noting happens.

I tried it on Firefox and Safari and that works perfectly fine still.

I used Chrome and tried the microphone feature on the homepage of google.com to do a search and that worked completely fine, although thats using the native feature of Chrome.

**Comment by aukkonen** [comment]
@DeanWronowski 
Okay, thank you! We'll keep trying to figure this out.

**Comment by aukkonen** [comment]
@DeanWronowski 
Just to make sure that some other site that uses the Web Audio API (that Chrome might not use on google.com.) works on your laptop, could you please try this demo:
https://mdn.github.io/voice-change-o-matic/

With "Voice setting" = Off and "Visualiser setting" = Frequency bars, you should see some bars jumping in the visualisation if the microphone works.

**Comment by DeanWronowski** [comment]
@aukkonen  - Yeah, that works fine. I can see the bars going up and down when talking.

**Comment by aukkonen** [comment]
@DeanWronowski 
We could check that Chrome is using an appropriate audio device in our demos. With the Flight booking demo (https://speechly-demos.herokuapp.com/flight-booking), could you first verify that it still is broken on Chrome, and then run the following in Console:
```
(await window.navigator.mediaDevices.getUserMedia({video: false, audio: {sampleRate: 16000, autoGainControl: true}})).getAudioTracks();
```
This should display the audio device (or, MediaStreamTrack) that is in use, and some information about its status. E.g., in my case the device is the MacBook Pro Microphone, and it's `readyState` value is set to "live".

**Comment by DeanWronowski** [comment]
@aukkonen 
See attachment for console debugging.  Hope it helps,
Also, the demo is still not displaying.

<img width="1116" alt="Screenshot 2021-11-15 at 14 48 35" src="https://user-images.githubusercontent.com/3969540/141802052-01739798-1ec1-42ff-9e6d-e16f486cba0d.png">
 any words when talking.

**Comment by DeanWronowski** [comment]
@aukkonen - Any update? Any closer to what might be causing it?

**Comment by aukkonen** [comment]
@DeanWronowski 
Sorry, nothing concrete as of yet. We are unable to reproduce this issue here.

Just to confirm: In your case the demo works on iMac (with same OS / Chrome version), but not on the laptop? (I just updated my Chrome to version 96.0.4664.45, btw.)

If you have the time to do some more debugging, you could try to get the [browser-client-example](https://github.com/speechly/speechly/tree/main/examples/browser-client-example) running locally. Basically you would have to clone this repo, set up the build tools (rush and pnpm), check out a debug branch, and then start the browser-client-example. As follows:
```
git clone git@github.com:speechly/speechly.git
cd speechly
npm install -g @microsoft/rush
npm install -g pnpm
git checkout test/browser-client-debug
rush update
rush build
cd examples/browser-client-example
rushx start
```
When the example application is running, open Console and click "Connect". You should see several "signalEnergy" values being printed on the Console, these should be nonzero. If the printed value is exactly equal to zero, then it seems that there is no signal coming in from the audio device.

You can also try the example according to its instructions. If you don't get a transcript, but the signalEnergy values are nonzero, then the problem is probably somewhere in between the audio device and the Websocket.

**Comment by DeanWronowski** [comment]
See attachment for console debug. Hope that helps,

<img width="1780" alt="Screenshot 2021-11-17 at 17 32 32" src="https://user-images.githubusercontent.com/3969540/142252078-427e7b57-4f73-4c70-be49-8739bec91488.png">

**Comment by aukkonen** [comment]
@DeanWronowski 
Thanks! It indeed looks like there is no audio coming in. But just to make sure, I made a small update to the debug branch with some more output. When you have the time, could you try pulling those changes, rebuilding the browser-client (rush build) and then trying the example again?

Also, on Chrome, if you open the page `chrome://media-internals/` in another window while running the example or one of our demos, you can see what Input Controllers (e.g. microphones) are active. What does this view show? (You can also click on an input controller to see more information about it.)

Thanks again!

**Comment by DeanWronowski** [comment]
@aukkonen 
When running the demo and then going to `chrome://media-internals/` it shows the following. Does that help?

<img width="1285" alt="Screenshot 2021-11-18 at 13 27 52" src="https://user-images.githubusercontent.com/3969540/142424144-f137b42a-b3a6-4d11-aa17-3ef12d474220.png">

**Comment by DeanWronowski** [comment]
On my iMac, it's got "default" in the device_id wonder if that's related?

**Comment by aukkonen** [comment]
@DeanWronowski 
Yes, this looks extremely interesting. And sort of weird, I must admit. A quick googling of the suspicious looking device_id (AudioDriverPluginDevice_UID) suggests that this might have something to do with Philips Hue Audio Sync App (that uses a virtual audio driver). Have you had something like this installed, or currently in use? Also, this might be visible in Audio Midi Setup as an audio device.

**Comment by DeanWronowski** [comment]
@aukkonen  - Amazing!
I uninstalled it about a year ago, but according to people online, it doesn't actually remove the virtual driver. I ran the following script and then it changed the default microphone back to the inbuilt one. No idea, why this suddenly changed when upgrading the OS. I couldn't change it from default to built-in on the site. The dropdown was disabled. 

https://github.com/maxrugen/huesync-remover/blob/main/remover.sh

Really appreciate your work.
